diff -ur scenario_original/ScenarioManager.cc scenario_modified/ScenarioManager.cc
--- scenario_original/ScenarioManager.cc	2021-05-06 11:12:46.000000000 +0200
+++ scenario_modified/ScenarioManager.cc	2020-07-15 04:18:34.000000000 +0200
@@ -308,13 +308,21 @@
     cModule *parentModule = getSimulation()->getSystemModule()->getModuleByPath(parentModulePath);
     if (parentModule == nullptr)
         throw cRuntimeError("Parent module '%s' is not found", parentModulePath);
-    cModule *submodule = parentModule->getSubmodule(submoduleName, 0);
-    bool vector = xmlutils::getAttributeBoolValue(node, ATTR_VECTOR, submodule != nullptr);
+
+    //TODO solution for inconsistent out-of-date vectorSize values in OMNeT++
+    int submoduleVectorSize = 0;
+    for (SubmoduleIterator it(parentModule); !it.end(); ++it) {
+        cModule *submodule = *it;
+        if (submodule->isVector() && submodule->isName(submoduleName)) {
+            if (submoduleVectorSize < submodule->getVectorSize())
+                submoduleVectorSize = submodule->getVectorSize();
+        }
+    }
+
+    bool vector = xmlutils::getAttributeBoolValue(node, ATTR_VECTOR, submoduleVectorSize > 0);
     cModule *module = nullptr;
     if (vector) {
-        cModule *submodule = parentModule->getSubmodule(submoduleName, 0);
-        int submoduleIndex = submodule == nullptr ? 0 : submodule->getVectorSize();
-        module = moduleType->create(submoduleName, parentModule, submoduleIndex + 1, submoduleIndex);
+        module = moduleType->create(submoduleName, parentModule, submoduleVectorSize + 1, submoduleVectorSize);
     }
     else {
         module = moduleType->create(submoduleName, parentModule);
